### DISCUSSION TITLE
osgText::Text is not updated

### DISCUSSION THREAD

**Discussion by okerstadt** [pelicanmapping/osgearth#2371] [Support Q&A] [answered]
Hi.
I have the actual master of osg and osgEarth and I'm trying to update the text of osg::Text. The first string is displayed but this never changes. For testing i used the code of osgearth_terrainprofile.cpp. At the end of the function redraw() the min/max labels are updated but the new values are not displayed. Can someone reproduce this or can someone give me a hint what the problem is here?

I get a lot of warnings when running the app of

Warning: detected OpenGL error 'invalid enumerant' at after RenderBin::draw(..)

If I delete line 227 "geode->addDrawable( geometry );" in createBackground the warnings are gone, so I think this is another problem...

Thanks
Okerstadt

**Comment by gwaldron** [comment, marked as answer]
Okerstadt,
If you're going to change `osgText::Text` after creation, you have to give it a dynamic data variance:
```c++
text->setDataVariance(osg::Object::DYNAMIC);
```
As for why the demo doesn't do that... probably because it's an ancient demo that was never updated when this became a requirement.
Hope this helps :)

**Reply by okerstadt** [reply]
Thank you Glenn, that helped. But now i give up migrating from version 2.10. There are too many things that are not working anymore. If one issue is solved i found two new problems...
